packages/react-server/src/ReactFlightServer.js

```diff
@@ -199,6 +202,8 @@ export type Request = {
   taintCleanupQueue: Array<string | bigint>,
   onError: (error: mixed) => ?string,
   onPostpone: (reason: string) => void,
+  // DEV-only
+  environmentName: string,
```

Comment:
Yeah personally I'd prefer we any cast the one place we read the name than do this to the Request type

Reply:
We have had this pattern elsewhere. It's nice because there's only one place to mess it up.

There will be many more places where `environmentName` gets used but the Request type is only initialized in one place.

Optional fields are bad for perf when you do mess them up because it breaks the hidden class. You could potentially accidentally add a field by writing to it unconditionally.

I don't like that we use optional fields for Fiber for example because we instead have bugs in the continued working with the code by doing undefined checks, which lets things pass and keep working even if they're undefined or missing. We typically check for example `=== null` exactly so that if we get an undefined flowing through the system it should error. Even if that errors only in production because then we'll quickly find the mistake.